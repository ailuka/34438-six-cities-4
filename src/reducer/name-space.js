const NameSpace = {
  ERROR: `ERROR`,
  FILTERS: `FILTERS`,
  OFFERS: `OFFERS`,
  USER: `USER`,
};

export {NameSpace};
