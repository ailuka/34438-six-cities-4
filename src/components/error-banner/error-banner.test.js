import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {NameSpace} from "../../reducer/name-space.js";
import ErrorBanner from "./error-banner.jsx";

const mockStore = configureStore([]);

describe(`ErrorBanner snapshot`, () => {
  it(`should not render`, () => {
    const store = mockStore({
      [NameSpace.ERROR]: {
        errorMessage: ``,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <ErrorBanner/>
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render`, () => {
    const store = mockStore({
      [NameSpace.ERROR]: {
        errorMessage: `New Error Text`,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <ErrorBanner/>
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
