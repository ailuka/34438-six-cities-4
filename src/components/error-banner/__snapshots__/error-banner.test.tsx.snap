// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBanner snapshot should not render 1`] = `null`;

exports[`ErrorBanner snapshot should render 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#e23939",
      "boxShadow": "0px 2px 14px 0px rgba(0,0,0,0.34)",
      "color": "#fff",
      "fontSize": "15px",
      "lineHeight": "15px",
      "padding": "5px",
      "position": "fixed",
      "textAlign": "center",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  New Error Text
</div>
`;
